### ISSUE TITLE
[SEVERE] Axis values not resolved correctly in report page tooltips when visual contains multiple measures

### ISSUE THREAD

**Issue by LiBerti13** [dm-p/powerbi-visuals-smlc#22]
I created a custom report page to be used as a tooltip with an SMLC visual. If I add only one measure into the Values field in the visual, the tooltip with my custom report page works fine. However, if I add two or more measures, the tooltip apparently picks up the wrong X axis value, and the tooltip contents are also based on this wrong value.

For example:
I have an an X axis between 0 and 100. With one measure on the SMLC visual, if I hover above e.g. 60, the tooltip correctly shows X=60, and the corresponding other values/measures from the tooltip page are also correct. However, if I have two measures then the tooltip shows X=30=60/2, and the other tooltip values are also based on X=30. If I have three measures on the SMLC visual then the tooltip shows X=20=60/3, etc.

This happens in both Power BI Desktop and Service (with Edge) under 64-bit Windows 10 Enterprise with SMLC version 2.1.0.130.

**Comment by dm-p** [comment]
Hi, and thanks for reporting - as mentioned on the other issue, do you have this replicated in a shareable workbook that you can email to me (daniel@coacervo.co)? It'll certainly help isolate the issue and test its resolution.

Report page tooltips are tricky as we don't have any control over their content; we delegate the current context to Power BI to handle them. As this is the same context as the regular tooltip (and if that's correct), this might require some questions from MS if it's not an obvious mistake on my side ;)

**Comment by LiBerti13** [comment]
Hi @dm-p! I've sent you an email with an example pbix. Perhaps the two bugs have a common cause so you could kill two flies with one slap!

**Comment by dm-p** [comment]
The bug here isn't the same as #23, but can be accomodated in the same piece of work ;)

Root cause is the selection ID (i.e. intersection of small multiple, axis and measure) was iterating through the data view incorrectly and was assigning the wrong index to the axis value, e.g.:

![image](https://user-images.githubusercontent.com/10572054/88858547-41b5c380-d24c-11ea-8d3c-0a9769c8ca79.png)
 So here, Measure = 2, Axis = 1 when it should be Axis = 2.

I've amended this to use the correct value of Axis when generating the selection ID, so this should work as expected in the next build, e.g.:

![issue_22_fix](https://user-images.githubusercontent.com/10572054/88859677-4d09ee80-d24e-11ea-8430-0c2e8f4afee1.gif)

In researching this I also realised that we are only sending the first measure context to Power BI for delegation (as we can only provide one context for canvas tooltips). I've updated the `selectionId` for each data point to include all measures for the current Small Multiple / Axis context, just in case.
